dds/DCPS/Serializer.cpp

```diff
@@ -49,10 +49,8 @@ Serializer::reset_alignment()
 void
 Serializer::smemcpy(char* to, const char* from, size_t n)
 {
-    (void) ACE_OS::memcpy(
-      reinterpret_cast<void*>(to),
-      reinterpret_cast<const void*>(from),
-      n);
+  OPENDDS_ASSERT(from);
```

Comment:
If !n, should this assert?  Does having an assert here really help?

Reply:
Eric ran into a problem with `from` being null from using a sequence inside a union. He changed it to throw an exception. Tim suggested I change it to an assert. I don't really care one way or another.